components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -23,12 +23,16 @@ import BrandedWallpaperLogo from '../../components/default/brandedWallpaper/logo
 
 // Helpers
 import VisibilityTimer from '../../helpers/visibilityTimer'
-import arrayMove from 'array-move'
-import { isGridSitePinned } from '../../helpers/newTabUtils'
 import { generateQRData } from '../../binance-utils'
 
+// API
+import {
+  customLinksEnabled,
+  isVisible,
+  setMostVisitedSettings
+} from '../../api/topSites'
```

Comment:
1) These names are confusing to read hundreds of lines down in the file. `isVisible` is not reflecting what it's related to. So either rename this to `shouldShowTopSites` or similar. Or `import * as TopSitesAPI` since we're not tree shaking. Or `import { isVisible as shouldShowTopSites } from '...'`.

2) If we're reading values direct from API and not subscribing to changes then what's making React re-render when the value changes? I don't see anything causing a re-render when `isVisible()` changes? If there is already a separate API event for changes that's already wired to redux (or internally in this react component to `component.setState`, which I can't see) then the value can be read from redux or react state.

Reply:
#1 is easy to fix; will push a fix here soon. #2 - good eye. Changing this value alone won't trigger a re-render. But any time the value changes in `InstantService`, there's a handler that fires which changes `gridSitesData` which I believe is what causes the re-render. Let me move the value into state and have it read from there

Reply:
RE: #2 interestingly enough, this value is being set properly. Not exactly sure where; trying to trace that down.

Reply:
OK fixed #1 and pushed; #2 I think gets re-rendered because the API does bind itself as a property. When grid site reducer changes the internal value it's storing (triggered by `brave\components\brave_new_tab_ui\apiEventsToStore.ts`), it must be triggering a reflow. Works great and I removed the unused newTabState.showTopSites. WDYT?

Reply:
It would never trigger a re-render when a non-stateful property changes, there is nothing that would re-call the render function. The render function is only called as a direct result of setState, which is called in the highest ancestor by redux when the redux state changes. That redux ancestor checks if our custom `mapStateToProps` function returns new data and, if it does, renders our first ancestor. Then only children down the tree are rendered normally only if any of their input props are different. If a component takes all of newTabData then it will definitely be rendered, which is why we want as much of the tree as possible to only consider simple properties it needs and nothing else (and also why it has to rely on incoming props and not some data from external API, unless it uses internal `this.setState` when that data changes).

In this case I think it's simple though. In Chromium c++ `MostVisitedInfoChanged` is called whenever `isVisible` or `customLinksEnabled` changes, which in your webui handler then fires `most-visited-info-changed` which you then forward in `apiEventsToStore` to redux which receives all the new data and can set it to state. I'll check if that's being handled in the reducer correctly but seems like since that's all wired up then you should just read `isVisible` and `customLinksEnabled` from redux state here, and not direct from the API module.

Reply:
FIXED